client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -122,11 +126,13 @@ public synchronized void setHosts(HttpHost... hosts) {
             throw new IllegalArgumentException("hosts must not be null nor empty");
         }
         Set<HttpHost> httpHosts = new HashSet<>();
+        AuthCache authCache = new BasicAuthCache();
```

Comment:
If we update the `AuthCache` rather than replace it, then it will potentially impact an ongoing request. I checked out the `BasicAuthCache` and `BasicScheme` code and it looks pretty lightweight?